lib/beaker/dsl/helpers/host_helpers.rb

```diff
@@ -224,6 +224,45 @@ def rsync_to host, from_path, to_path, opts = {}
           end
         end
 
+        # Copy a remote file to the local system and save it under a directory
+        # meant for storing SUT files to be viewed in the event of test failures.
+        #
+        # Files are stored locally with the following structure:
+        #   ./<archive_root>/<hostname><from_path>
+        #
+        # This can be used during the post-suite phase to persist relevant log
+        # files from the SUTs so they can available with the test results
+        # (without having to preserve the SUT host and SSH in after the fact).
+        #
+        # Example
+        #
+        #   Archive the Puppet Server log file from the master ('abc123'),
+        #   and archive the Puppet Agent log file from the agent ('xyz098'):
+        #
+        #     archive_file_from(master, '/var/log/puppetlabs/puppetserver.log')
+        #     archive_file_from(agent, '/var/log/puppetlabs/puppetagent.log')
+        #
+        #   Results in the following files on the test runner:
+        #
+        #     archive/sut-files/abc123/var/log/puppetlabs/puppetserver.log
+        #     archive/sut-files/xyz098/var/log/puppetlabs/puppetagent.log
+        #
+        # @param [Host] host A host object (or some object that can be passed to
+        #                    #scp_from)
+        # @param [String] from_path A remote absolute path on the host to copy.
+        # @!macro common_opts
+        # @option [String] archive_root The local directory to store the copied
+        #                               file under. Defaults to
+        #                               'archive/sut-files'.
+        #
+        # @return [Result] Returns the result of the #scp_from operation.
+        def archive_file_from(host, from_path, opts = {}, archive_root = 'archive/sut-files')
```

Comment:
In my opinion for something like this it's better to have users "configure" the behavior where it is used rather than someplace far away (is that what "action at a distance means"?...when some configuration option set somewhere else configures the behavior for a localized piece of code?). But I'm definitely not a Beaker expert so I'll defer to pretty much anyone else's opinion.